src/screens/Main.tsx

```diff
@@ -1,8 +1,13 @@
-import { useInfiniteQuery } from "@tanstack/react-query";
+import { useInfiniteQuery, useQuery } from "@tanstack/react-query";
+import { useLocation } from "react-router-dom";
 import { Product, getProductsQuery } from "../api/products";
 import { ProductCard, icons } from "../components";
 
 export function Main() {
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const searchTerm = searchParams.get("search") || "";
```

Comment:
You should use `??` instead of `||` in this case.

https://stackoverflow.com/questions/61480993/when-should-i-use-nullish-coalescing-vs-logical-or